### ISSUE TITLE
Feature Mismatch Between Map and Rendered Features Query

### ISSUE THREAD

**Issue by mcollins-GPI** [mapbox/mapbox-gl-js#13297]
I have listed this potential bug on [Github](https://stackoverflow.com/questions/79051946/mapbox-feature-mismatch-between-map-and-rendered-features-query) as well, but figured this might be the more appropriate place to ask the question. I am getting strange behavior with Mapbox 3 when I click on the map. It seems like the features arbitrarily correspond (or don't) with the location of the map click. In the example below, I am clicking in empty space and getting back 3 features:

[![Map click event where no features should be captured][1]][1]

In the second image I am displaying the coordinates of the features selected and they are not very close (when compared to the scale at zoom level 16) to the point clicked. The point clicked is the second command shown:

[![Queried features vs Map Click Point @ Zoom Level 16][2]][2]

If I then query the features that are returned in Postgres to see the actual locations being selected, I get even another representation of the data:

[![Representation of mapped data from database][3]][3]

To put it simply, something is not right here, and I am not sure why the click event would return data that is not in the same location, or in some cases even close. While in others it returns the feature displayed on the map correctly. Any guidance or starting steps to break this problem down would be most appreciated.

  [1]: https://i.sstatic.net/4KYUR2Lj.png
  [2]: https://i.sstatic.net/f5tU2l56.png
  [3]: https://i.sstatic.net/MXKBFhpB.png

**Comment by stepankuzmin** [comment]
Hi @mcollins-GPI,

Thank you for opening this ticket. Please provide a minimal, complete, verifiable demonstration of the issue to help the maintainers and community understand and address the problem accurately. We recommend using https://jsbin.com/. Please include information about the browser version and operating system used.
